app/build.gradle

```diff
@@ -218,10 +218,10 @@ dependencies {
     // See https://github.com/mozilla/application-services/blob/master/.buildconfig-android.yml#L1 for latest version
     testImplementation "org.mozilla.appservices:full-megazord-forUnitTests:0.37.1"
 
-    compileOnly files('libs/amazon-device-messaging-1.0.1.jar')
-    testImplementation files('libs/amazon-device-messaging-1.0.1.jar')
-    // See https://github.com/mozilla/application-services/blob/master/.buildconfig-android.yml#L1 for latest version
-    testImplementation "org.mozilla.appservices:full-megazord-forUnitTests:0.37.1"
+    // Download from https://developer.amazon.com/sdk-download
+    // This library has not been updated since 2013, so it might be unlikely that the version will change
+    compileOnly files("libs/amazon-device-messaging-${adm_version}.jar")
+    testImplementation files("libs/amazon-device-messaging-${adm_version}.jar")
```

Comment:
I'm not sure what's best here. Now that we fixed the startup crash, we might be able to change this to `testCompileOnly` and remove the shadows. However, if we ever try to run tests that use these classes, the tests might crash without the shadows and the runtime dependencies.

Any thoughts? Maybe it's better just to add a comment and leave it how it is because it works now and we can go add more user value. :)

If you think it's appropriate to change this, can you file a bug for someone to change this into a-c, which also uses `testImplementation`?

Reply:
I think I'm confused how this relates to the startup crash. If the shadows are removed, aren't the tests going to fail on every PR?

Reply:
> If the shadows are removed, aren't the tests going to fail on every PR?

I'm not sure – we can try it! :)

My thinking was that Robolectric crashes for the same reason the app crashes on startup (it tries to load classes whose implementations are not available) and so by fixing the startup crash, we fixed the issue, provided those APIs are not accessed. However, it could be that's not the case (e.g. Robolectric has additional startup logic around class loading that will crash even though the app doesn't) and we can't remove the shadows.

Then again, we may need to load those Amazon classes in testing, even if it's just to mock them, so maybe it doesn't make sense to remove the classes at all. I'm not sure. Sorry if this is unclear but we can talk on Zoom to work through it if you'd like.